### ISSUE TITLE
How can I control where compiled lib and bins are placed ?

### ISSUE THREAD

**Issue by kobi-ca** [conan-io/conan#7944]
Hello,

I created an example using conan new etc... and when building lib, bin and gtest bins, are placed in bin/ lib/ dirs.
This is happening even w/o running `make install`.
 For example, gtest bins would be under `<build-dir>/.../tests/` mirroring source dir.
How can I control where libs/bins are placed?

thanks

**Comment by memsharded** [comment]
Hi @kobi-ca 

The question is not fully clear, could you please specify details, recipes, commands of what you mean, and what you want to achieve? How do you get the ``<build-dir>/.../tests/`` folder? Which gtest recipe? Why ``make install``? Why ``gtest`` if you created a ``conan new`` library example, that is not connected to gtest at all?

Please, try to provide the accurate, detailed steps to reproduce what you mean. Thanks!

**Comment by kobi-ca** [comment]
Hi @memsharded . sorry I was not clear.
Let me try again.
I was using this is a reference 
https://levelup.gitconnected.com/conan-package-manager-tutorial-bfb308ddd7a
and developed my own lib on top of it.

In general, when you have a cmakelists.txt and it defined a lib or executable, the directory where the lib or exe are built, is the same as the source dir structure, but just under <build>.

e.g.
```
root-
     - a/
     - ---lib/
     - b/
     - ---exe/
    build
         - a/
         - ---lib/
         - b/
         - ---exe/
```

As you can see the cmake structure under <build> dir is the same as the source one. This is how it works in cmake.
So my exe will be under `build/b/exe/` and lib will be under `build/a/lib`

with the default generated conanfile.py - you can see it in the link I posted above - the lib and exe will go under `<build>/lib` and `<build>/exe`.
I gave a gtest just an additional example.
in a regular cmake structure the gtest executable will be under `<build>/path/mirrored/from/src` but instead with this conanfile.py it goes to `<build>/bin/`

So my question is - what in conan cmake creates this behavior? i.e. changing the build path to be under `<build>/lib` or `<build>/bin` instead of  `<build>/path/mirrored/from/src` 

Hope this is cleaner.

thanks.

**Comment by memsharded** [comment]
I think now I understand. The ``conan_basic_setup()`` cmake function is calling ``conan_output_dirs_setup()`` internally, which defines the default CMake "output" directories.

If you pass the argument ``conan_basic_setup(NO_OUTPUT_DIRS)`` it will not redefined the output folders. You can read more about it in: https://docs.conan.io/en/latest/reference/generators/cmake.html#macros-available-in-conanbuildinfo-cmake

Please try that and let me know.


By the way, the new integrations like ``cmake_find_package_multi`` and the toolchains, will not define by default this behavior.

**Comment by kobi-ca** [comment]
perfect! let me try this. many thanks @memsharded

**Comment by kobi-ca** [comment]
Worked! thanks for the pointers.
